src/main/java/com/google/sps/server/CalendarUtils.java

```diff
@@ -14,105 +14,85 @@
 
 package com.google.sps.server;
 
-import java.util.Calendar;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+
+import com.google.api.services.calendar.Calendar;
+import com.google.api.services.calendar.model.CalendarListEntry;
+
 import com.google.sps.data.*;
 
 public class CalendarUtils {
    /**
-   * Helper function to calculate the day of the month of the next Sunday,
-   * to know when to start the schedule. 
+   * Helper function to get the date for the calendar event
    *
-   * @param calendar an instance of Calendar, so can know the current date
-   * @return an int representing the date of next Sunday
+   * @param startDate the start date of the user's college term
+   * @param timeInMin the time, in minutes of the week, for an event (either the start or end time) 
+   * @param courseDayOfWeek the day of week the current course section starts on 
+   * @return a LocalDate representing the date to put the course on the calendar
    */
-  public static int getDateOfNextSunday(Calendar calendar) {
-    int currentDay = calendar.get(Calendar.DAY_OF_MONTH);
-    int currentDayOfWeek = calendar.get(Calendar.DAY_OF_WEEK);
-    int nextSun = currentDay + 8 - currentDayOfWeek;
-    return nextSun;
+  private static LocalDate getCourseDate(LocalDate startDate, int timeInMin, int courseDayOfWeek) {
+    int startDateDayOfWeek = startDate.getDayOfWeek().getValue();
+    long daysToAdd = 0;
+    if (courseDayOfWeek > startDateDayOfWeek) {
+      daysToAdd = courseDayOfWeek - startDateDayOfWeek;
+    } else {
+      daysToAdd = courseDayOfWeek + 7 - startDateDayOfWeek;
+    }
+    LocalDate courseDate = startDate.plusDays(daysToAdd);
+    return courseDate;
   }
 
    /**
-   * Helper function to know the current month. 
+   * Helper function to get the time (start or end) for the calendar event
    *
-   * @param calendar an instance of Calendar, so can know the current date
-   * @return an int representing the month, adjusted for DateTime's formatting
+   * @param timeInMin the time, in minutes of the week, for an event (either the start or end time) 
+   * @param courseDayOfWeek the day of week the current course section starts on 
+   * @return a LocalTime representing the time for creating the calendar event for the current course
    */
-  public static int getCurrentMonth(Calendar calendar) {
-    return calendar.get(Calendar.MONTH) + 1;
-  }
-  
+  private static LocalTime getCourseTime(int timeInMin, int courseDayOfWeek) {
+    int dayInHours = courseDayOfWeek * 24;
+    int hour = (int) Math.floor((timeInMin / 60) - dayInHours);
+    int minute = (int) (60 * (((timeInMin / 60.0) - dayInHours) - hour));
```

Comment:
Would this achieve the same thing?

```
int minute = timeInMin % 60;
```

Reply:
yes - that should be the same, not sure how I ended up with such a complicated equation but I definitely should have checked to simplify it. Sorry about that and thanks for noticing!